PWGJE/Tasks/jetChCorr.cxx

```diff
@@ -0,0 +1,454 @@
+              }
+              if (ch_l != ch_nl) {
+                registry.fill(HIST("hr2_ch_d_pt"), ptJet);
+                registry.fill(HIST("hr2_ch_d_kt"), kt_p);
+                registry.fill(HIST("hr2_ch_d_z"), z);
+                registry.fill(HIST("hr2_ch_d_ft"), log10(fT));
+                registry.fill(HIST("hr2_ch_d_mult"), ch_mult);
+                registry.fill(HIST("hr2_ch_d_zg"), zg);
+                registry.fill(HIST("hr2_ch_d_rg"), log(rg));
+                registry.fill(HIST("hr2_ch_d_lu"), log(1 / rg), log(kt_p));
+              }
+            }
+            ////////  for 3rd marched split
+            if (nsd >= 2) {
+              if (ch_l == ch_nl) {
+                registry.fill(HIST("hr3_ch_s_pt"), ptJet);
+                registry.fill(HIST("hr3_ch_s_kt"), kt_p);
+                registry.fill(HIST("hr3_ch_s_z"), z);
+                registry.fill(HIST("hr3_ch_s_ft"), log10(fT));
+                registry.fill(HIST("hr3_ch_s_mult"), ch_mult);
+                registry.fill(HIST("hr3_ch_s_zg"), zg);
+                registry.fill(HIST("hr3_ch_s_rg"), log(rg));
+                registry.fill(HIST("hr3_ch_s_lu"), log(1 / rg), log(kt_p));
+              }
+              if (ch_l != ch_nl) {
+                registry.fill(HIST("hr3_ch_d_pt"), ptJet);
+                registry.fill(HIST("hr3_ch_d_kt"), kt_p);
+                registry.fill(HIST("hr3_ch_d_z"), z);
+                registry.fill(HIST("hr3_ch_d_ft"), log10(fT));
+                registry.fill(HIST("hr3_ch_d_mult"), ch_mult);
+                registry.fill(HIST("hr3_ch_d_zg"), zg);
+                registry.fill(HIST("hr3_ch_d_rg"), log(rg));
+                registry.fill(HIST("hr3_ch_d_lu"), log(1 / rg), log(kt_p));
+              }
+            }
+          }
+          break;
+        }
+        nsd++;
+      }
+      daughterSubJet = parentSubJet1;
+    }
+    if constexpr (!isSubtracted && !isMCP) {
+      registry.fill(HIST("h2_jet_pt_jet_nsd"), jet.pt(), nsd);
+    }
+    if constexpr (!isSubtracted && isMCP) {
+      registry.fill(HIST("h2_jet_pt_part_jet_nsd_part"), jet.pt(), nsd);
+    }
+    if constexpr (isSubtracted && !isMCP) {
+      registry.fill(HIST("h2_jet_pt_jet_nsd_eventwiseconstituentsubtracted"), jet.pt(), nsd);
+    }
+    // outputTable(energyMotherVec, ptLeadingVec, ptSubLeadingVec, thetaVec);
+  }
+
+  //  template <bool isSubtracted, typename T, typename U, typename V>
+  //  void analyseCharged(T const& jet, U const& tracks, V& outputTable)
+  template <bool isSubtracted, typename T, typename U>
+  void analyseCharged(T const& jet, U const& tracks)
+  {
+    jetConstituents.clear();
+
+    int nn = 0;
+    int iord[50];
+    float ptc[50];
+    for (auto& jetConstituent : jet.template tracks_as<aod::JTracks>()) {
+      TVector3 p_leading(jetConstituent.px(), jetConstituent.py(), jetConstituent.pz());
+      ptc[nn] = p_leading.Perp();
+      nn++;
+    }
+    TMath::Sort(nn, ptc, iord);
```

Comment:
is it not more efficient than having 2 for loops and this sorting algorithm to just have one for loop which does something like:
int pTL = 0.0
int pTNL = 0.0
for loop over jet constituents{

constituentPt = jetConstituent.pt()
if (constituentPt > pTL){
pTNL = pTL
pTL = constituentPt
set all the leading and subleading constituent info you need
}
else if (constituentPt > pTNL){
pTNL = constituentPt
Set all the pTNL values you need
}
}

Reply:
It need to keep the second loop since I need to know leading and next leading particles before starting calculations;  unless cam jump to leading and n-leading track without loop.